tests/qos/test_buffer.py

```diff
@@ -1227,59 +1263,151 @@ def test_exceeding_headroom(duthosts, rand_one_dut_hostname, conn_graph_facts, p
 
+        # Update static profile to a larger size, which makes it exceeds the port headroom limit
+        # Setup the log analyzer
+        loganalyzer, marker = init_log_analyzer(duthost,
+                                                'Configure a larger size to a static profile',
+                                                ['BUFFER_PROFILE .* cannot be updated because .* referencing it violates the resource limitation',
+                                                 'Unable to update profile for port .*. Accumulative headroom size exceeds limit'])
+
         logging.info('[Update headroom override to a larger size]')
-        duthost.shell('config buffer profile set test-headroom --xon 18432 --xoff 860160 -headroom 878592')
+        duthost.shell('config buffer profile set test-headroom --size {}'.format(int(maximum_profile['size']) * 2))
 
         # This should make it exceed the limit, so the profile should not applied to the APPL_DB
+        time.sleep(20)
         size_in_appldb = duthost.shell('redis-cli hget "BUFFER_PROFILE_TABLE:test-headroom" size')['stdout']
-        pytest_assert(size_in_appldb == '68432', 'The profile with a large size was applied to APPL_DB, which can make headroom exceeding')
+        pytest_assert(size_in_appldb == maximum_profile['size'], 'The profile with a large size was applied to APPL_DB, which can make headroom exceeding')
+
+        # Check log
+        check_log_analyzer(loganalyzer, marker)
+
+        # Restore config
         duthost.shell('config interface buffer priority-group lossless set {} {}'.format(port_to_test, '3-4'))
         duthost.shell('config buffer profile remove test-headroom')
-        logging.info('[Clean up]')
+
+        # 4. Configure a long cable length with shared headroom pool enabled.
+        loganalyzer, marker = init_log_analyzer(duthost,
+                                                'Toggle shared headroom pool',
+                                                ['BUFFER_PROFILE .* cannot be updated because .* referencing it violates the resource limitation',
+                                                 'Unable to update profile for port .*. Accumulative headroom size exceeds limit',
+                                                 'refreshSharedHeadroomPool: Failed to update buffer profile .* when toggle shared headroom pool'])
+
+        # Enable shared headroom pool
+        duthost.shell('config buffer shared-headroom-pool over-subscribe-ratio 2')
+        time.sleep(20)
+        # And then configure the cable length which causes the accumulative headroom exceed the limit
+        duthost.shell('config interface cable-length {} {}m'.format(port_to_test, violating_cable_length))
+        expected_profile = 'pg_lossless_{}_{}m_profile'.format(original_speed, violating_cable_length)
+        check_pg_profile(duthost, 'BUFFER_PG_TABLE:{}:3-4'.format(port_to_test), expected_profile)
```

Comment:
capture the return status and assert accordingly

Reply:
The same. The test will fail if an error occurs.